include/wx/private/extfield.h

```diff
@@ -0,0 +1,84 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/private/extfield.h
+// Purpose:     Declare wxExternalField helper
+// Author:      Vadim Zeitlin
+// Created:     2017-11-21
+// Copyright:   (c) 2017 Vadim Zeitlin <vadim@wxwidgets.org>
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_PRIVATE_EXTFIELD_H_
+#define _WX_PRIVATE_EXTFIELD_H_
+
+// ----------------------------------------------------------------------------
+// wxExternalField: store object data outside of it
+// ----------------------------------------------------------------------------
+
+// This class allows to store some data without consuming space for the objects
+// that don't need it and can be useful for avoiding to add rarely used fields
+// to the classes that are used by many objects, e.g. wxWindow.
+//
+// Note that using this class costs in speed and convenience of access to the
+// field, which requires a hash lookup instead of accessing it directly. It
+// also only currently works for heap-allocated fields as it's probably never
+// worth using it for fields of simple types.
+//
+// Template parameter Object is the class that "contains" the field, Field is
+// the type of the field itself and FieldMap is the hash map, defined
+// separately using WX_DECLARE_HASH_MAP(), with Object* as the key and Field*
+// as the value type.
+template <typename Object, typename Field, typename FieldMap>
+class wxExternalField
+{
+public:
+    typedef Object ObjectType;
+    typedef Field FieldType;
+    typedef FieldMap MapType;
```

Comment:
Anything to ensure, that the map is thread safe?

Reply:
No, as mentioned in the comment, this is mostly meant to be used by `wxWindow`, i.e. from the GUI thread only.